src/start_aws_gha_runner/start.py

```diff
@@ -112,12 +121,26 @@ def _build_user_data(self, **kwargs) -> str:
             "templates/user-script.sh.templ"
         )
         with template.open() as f:
-            template = f.read()
-            try:
-                parsed = Template(template)
-                return parsed.substitute(**kwargs)
-            except Exception as e:
-                raise Exception(f"Error parsing user data template: {e}")
+            template_content = f.read()
+
+        try:
+            parsed = Template(template_content)
+            runner_script = parsed.substitute(**kwargs)
+
+            # If custom userdata is provided, prepend it to the runner script
+            if self.userdata:
+                combined_script = f"""#!/bin/bash
+# Custom user data script
+{self.userdata}
+
+# Runner setup script
+{runner_script.replace('#!/bin/bash', '').strip()}
```

Comment:
🐑 maybe slightly nicer to have this be a substitution passed in the template instantiation above? (Or just leave it, presumably it would do no harm?) I worry about the runner script including embedded bash scripts.

Reply:
Good point, done!